File: ALGORITHM TOOLBOX/02_greedy_algorithms_starter_files/dot_product/dot_product.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

/*bool comp(int i, int j){
  return (i>j);
}
*/
long long min_dot_product(vector<int> a, vector<int> b) {
  if(!a.size()){
    return  0;
  }
  sort(a.begin(), a.end());
  sort(b.begin(), b.end());
  long long result = 0;
  for (size_t i = 0; i < a.size(); i++) {
    result += a[i] * b[i];
  }
  return result;
}

int main() {
  size_t n;
  cin >> n;
  vector<int> a(n), b(n);
  for (size_t i = 0; i < n; i++) {
    cin >> a[i];
  }
  for (size_t i = 0; i < n; i++) {
    cin >> b[i];
  }
  cout << min_dot_product(a, b) << endl;
}

File: ALGORITHM TOOLBOX/02_greedy_algorithms_starter_files/dot_product/DotProduct.java
import java.util.Arrays;
import java.util.Scanner;


public class DotProduct {

    private static long minDotProduct(int[] a, int[] b) {
        Arrays.sort(a);
        Arrays.sort(b);
        long result = 0;
        for (int i = 0, n = a.length; i < a.length; i++) {
            result += a[i] * b[n - 1 - i];
        }
        return result;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] a = new int[n];
        for (int i = 0; i < n; i++) {
            a[i] = scanner.nextInt();
        }
        int[] b = new int[n];
        for (int i = 0; i < n; i++) {
            b[i] = scanner.nextInt();
        }
        System.out.println(minDotProduct(a, b));
        scanner.close();
    }
}
